nats.go

```diff
@@ -4860,6 +4984,12 @@ func (s *Subscription) validateNextMsgState(pullSubInternal bool) error {
 	if s.mcb != nil {
 		return ErrSyncSubRequired
 	}
+	// if this subscription previously had a permissions error
+	// and no reconnect has been attempted, return the permissions error
+	// since the subscription does not exist on the server
+	if s.conn.Opts.PermissionErrOnSubscribe && s.permissionsErr != nil {
```

Comment:
We won't really terminate the subscription, but rather return a permission error on each `Next()` call until there is a reconnect attempt, at which point client will attempt to recreate them.

Without this option, it works pretty much the same only `Next()` returns timeout since the sub does not exist on the server any more.

Reply:
hm, techcnically, server could also reload permissions without restart, and we would still be stuck in the loop?

Reply:
Server can reload permissions, but that does not matter since it already removed the subscription in the first place after permission error right?

And only after reconnect we try to recreate it.